peachjam/resources.py

```diff
@@ -38,54 +38,93 @@
 logger = logging.getLogger(__name__)
 
 
-class JurisdictionWidget(ForeignKeyWidget):
+class CharRequiredWidget(CharWidget):
+    def __init__(self, field):

```

Comment:
Minor: you could turn this required functionality into a mixin, since it's repeated in about 4 places.

Reply:
Had the same thought but the issue is the implementation is slightly different for ForeignKeyWidget vs CharWidget, and other is a CustomWidget. :grimacing: